steamclog/src/main/java/com/steamclock/steamclog/Destinations.kt

```diff
@@ -145,49 +163,45 @@ internal class ConsoleDestination: Timber.DebugTree() {
-    private var logTimestampFormat = "yyyy-MM-dd'.'HH:mm:ss.SSS"
     private var fileExt = "txt"
 
     override fun isLoggable(priority: Int): Boolean {
         return isLoggable(SteamcLog.config.logLevel.file, priority)
     }
 
-    //---------------------------------------------
-    // Reformats console output to include file and line number to log.
-    //---------------------------------------------
-    override fun createStackElementTag(element: StackTraceElement): String? {
-        return "(${element.fileName}:${element.lineNumber}):${element.methodName}"
-    }
-
     //---------------------------------------------
     // Allows us to print out to an external file if desired.
     //---------------------------------------------
     override fun log(priority: Int, tag: String?, message: String, throwable: Throwable?) {
-        val wrapper = throwable as? SteamclogThrowableWrapper
-        val originalMessage = wrapper?.originalMessage ?: message
+        val wrapper = SteamclogThrowableWrapper.from(throwable)
         val originalThrowable = wrapper?.originalThrowable
-        val extraData = wrapper?.extraData?.let { ": $it" } ?: run { "" }
-        val fullMessage = "$originalMessage$extraData"
 
-        printLogToExternalFile(priority, tag, fullMessage)
+        val fullMessage = generateFullLogMessage(
+            priority,
+            createCustomStackElementTag(),
+            includeTimestamp = true,
+            includeEmoji = false,
+            throwable,
+            message
+        )
+
+        printLogToExternalFile(fullMessage)
         originalThrowable?.let {
-            printLogToExternalFile(priority, tag, Log.getStackTraceString(it))
+            printLogToExternalFile(Log.getStackTraceString(it))
         }
     }
 
     //---------------------------------------------
     // Support to write logs out to External HTML file.
     //---------------------------------------------
-    private fun printLogToExternalFile(priority: Int, tag: String?, message: String) {
+    /**
+     * [EMOJI?] [Date/Time UTC?] [App ID[Process ID?: Thread ID?]] ...
+     * [Log Level] [Thread name] [(FileName.ext:line number): functionName()] > ...
+     * [The actual log message], JSON: [JSON Object if applicable]
+     */
+    private fun printLogToExternalFile(message: String) {
         try {
-            val date = Date()
-            val logTimeStamp = SimpleDateFormat(logTimestampFormat, Locale.US).format(date)
-            val appId = BuildConfig.LIBRARY_PACKAGE_NAME
-            val processId = android.os.Process.myPid()
-            val threadName = Thread.currentThread().name
-            val logStr = "$logTimeStamp $appId[$processId:$threadName] [$priority] [$tag] > $message \r\n"
-            // If file created or exists save logs
-            getExternalFile()?.let { file -> file.appendText(logStr) }
+            getExternalFile()?.let { file -> file.appendText("$message\r\n") }
```

Comment:
Looks like this wasn't added in this PR, but do you know if this will automatically close the file after writing is finished?

Reply:
Hrm. Doesn't look like I'm closing it, but it is still working. I've created https://github.com/steamclock/steamclog-android/issues/58 to follow up on it and make sure it's handled correctly.